### PR TITLE
Reworked StructRenameRefatoring to use new search

### PR DESCRIPTION
**Pull request by azoitl**
With this change the struct type rename refactoring uses the new search to find the correct usages of types. Furthermore, for dependent changes (i.e., struct in struct usage) the right type entry is used.

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/StructuredTypeMemberChange.java [hunk 1]
```diff
@@ -62,8 +59,8 @@ public Change perform(final IProgressMonitor pm) throws CoreException {
 
 		final StructuredType structuredTypeEditable = (StructuredType) affectedStruct.getTypeEntry().getTypeEditable();
 		for (final VarDeclaration varDeclaration : structuredTypeEditable.getMemberVariables()) {
-			final String typeName = varDeclaration.getTypeName();
-			if (typeName.equals(oldName)) {
+			if ((varDeclaration.getType() != null && typeEntry == varDeclaration.getType().getTypeEntry())
+					|| (typeEntry.getFullTypeName().equals(varDeclaration.getFullTypeName()))) {
```
Comment (pazi146) [inline plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/StructuredTypeMemberChange.java]: varDeclaration.getFullTypeName() won't give the full name if its package is imported on that location - only TypeEntry.getFullTypeName() is guaranteed to return the fully qualified name

Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/StructuredTypeMemberChange.java]: ok then I think it is better to remove the or part. Will do that.

**Comment by oberlehner** [comment]
Using the struct to be renamed in the interface of a simple FB, which is also used in  results in a deadlock after the rename.
before:
image


Preview (strangely the StructUser Simple has not be found in the search):
image


Afterwards everything hangs:

image

**Comment by azoitl** [comment]
@oberlehner thx for finding this. I found a bug in the DataTypesearch (stupid copy paste error) and I think I solved the deadlock.

**Pull request merged**